### PR TITLE
feat(api): Track params for RPC methods

### PR DESCRIPTION
**Pull request by slowli**
## What ❔

Tracks params for the RPC method calls in the metadata middleware, so that they can be reported for oversized responses, dropped client calls etc.

## Why ❔

Would allow to debug performance issues more easily.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.
- [x] Spellcheck has been run via `zk spellcheck`.
- [x] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

File: core/lib/zksync_core/src/api_server/web3/backend_jsonrpsee/mod.rs [hunk 1]
```diff
@@ -19,6 +21,54 @@ pub mod namespaces;
...
+/// to another `RawParams` with the same lifetime.
+///
+/// # Why?
+///
+/// We need all this complexity because we'd like to access request params in a generic way without an overhead
+/// (i.e., cloning strings for each request – which would hit performance and fragment the heap).
+/// One could think that `jsonrpsee` should borrow params by default;
+/// if that were the case, we could opportunistically expect `RawParams` to be `None` or `Some(Cow::Borrowed(_))`
+/// and just copy `&RawValue` in the latter case. In practice, `RawParams` are *never* `Some(Cow::Borrowed(_))`
+/// because of a bug (?) with deserializing `Cow<'_, RawValue>`: https://github.com/serde-rs/json/issues/1076
```
Comment (slowli) [inline core/lib/zksync_core/src/api_server/web3/backend_jsonrpsee/mod.rs:36]: Here's motivation behind this design. I guess we could try fixing the bug (?) with `Cow<'_, RawValue>`, but I'm not sure where to even start with this: is it really a `serde_json` / `serde` bug? Or would it make sense to somehow fix deserialization in `jsonrpsee` (e.g., with manually implemented deserialization logic)?

**Review by slowli — COMMENTED** [review]
I'm fairly sure that the introduced `unsafe` code is sound, but not 100% sure. So I won't be particularly disappointed if we decide not to merge this PR.

**Review by RomanBrodetski — APPROVED** [review]

File: core/lib/zksync_core/src/api_server/web3/backend_jsonrpsee/middleware.rs [hunk 2]
```diff
@@ -145,27 +147,42 @@ where
...
     }
 }
 
-impl<F: Future<Output = MethodResponse>> Future for WithMethodCall<F> {
+impl<'a, F> WithMethodCall<'a, F> {
+    fn new(inner: F, call: MethodCall<'a>) -> Self {
+        // Wrap a call into a span with unique correlation ID, so that events occurring in the span can be easily filtered.
+        // This works as a cheap alternative to Open Telemetry tracing with its trace / span IDs.
+        let call_span =
+            tracing::debug_span!("rpc_call", correlation_id = generate_correlation_id());
```
Comment (slowli) [inline core/lib/zksync_core/src/api_server/web3/backend_jsonrpsee/middleware.rs:171]: I've used this trick on my previous job (although there, it worked with conjunction with Jaeger tracing), and it really helped to investigate some errors. It is another allocation per each request, but it's small.

**Review by RomanBrodetski — DISMISSED** [review]
Love it!

**Review by RomanBrodetski — APPROVED** [review]

**Pull request merged**